backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/ProcessDownVmCommand.java

```diff
@@ -155,7 +158,18 @@ protected void executeCommand() {
             Guid alternativeHostsList = vmHasDirectPassthroughDevices ? getVm().getDedicatedVmForVdsList().get(0) : null;
             refreshHostIfNeeded(hostId == null ? alternativeHostsList : hostId);
         }
-
+        VdsManager vdsManager = resourceManager.getVdsManager(getParameters().getHostId());
+        String pinning = getVm().getCpuPinningPolicy() == CpuPinningPolicy.MANUAL ? getVm().getCpuPinning() : getVm().getCurrentCpuPinning();
+        if (pinning != null && !pinning.isBlank()) {
+            synchronized (vdsManager) {
+                vdsManager.getCpuTopology().stream().filter(cpu -> CpuPinningHelper.getAllPinnedPCpus(getVm().getCurrentCpuPinning()).contains(cpu.getCpu())).forEach(cpu -> cpu.setVmId(null));
+                vdsManager.updateDynamicToCachedVds();
+            }
+        }
```

Comment:
why do we need this block if we change all the cpus that are assigned to this vm below?